cafebabel/static/js/edit-article.js

```diff
@@ -0,0 +1,146 @@
+class Tags {
+  constructor() {
+    const domContext = document.querySelector('.tags')
+    this.list = domContext.querySelector('.tags-list')
+    this.values = Array.from(this.list.querySelectorAll('input')).map(
+      input => input.value
+    )
+  }
+  static addTagsRemoveListener() {
+    const tagsButtonRemove = document.querySelectorAll('.tags .tags-list li a')
+    tagsButtonRemove.forEach(tagButtonRemove =>
+      Listener.clickRemove(tagButtonRemove)
+    )
+  }
+  checkSubmission(query) {
+    return query && !this._isTag(query)
+  }
+  add(query) {
+    this.values = this.values.concat(query)
+  }
+  remove(query) {
+    this.values = this.values.filter(value => value !== query)
+  }
+  display() {
+    const container = this.list.cloneNode(false)
+    this.list.replaceWith(this._createTagsList(container, this.values))
+    Tags.addTagsRemoveListener()
+  }
+  _isTag(tag) {
+    return this.values.includes(tag)
+  }
+  _createTagsList(container, tagValues) {
+    tagValues.forEach((tagValue, index) =>
+      container.appendChild(this._createTag(tagValue, index))
+    )
+
+    return container
+  }
+  _createTag(tagValue, index) {
+    const li = document.createElement('li')
+    const input = document.createElement('input')
+    const a = document.createElement('a')
+    input.type = 'hidden'
+    input.name = `tag-${index + 1}`
+    input.setAttribute('list', 'tags')
+    input.value = tagValue
+    li.append(tagValue)
+    li.append(input)
+    li.append(a)
+
+    return li
+  }
+}
+
+class Effect {
+  static fadeIn(element) {
+    element.style.opacity = 0
+    ;(function fade() {
```

Comment:
Out of curiosity, why IIFEs here and below?

Reply:
Not fine with the old version